packages/hoppscotch-backend/src/auth/auth.module.ts

```diff
@@ -26,9 +27,9 @@ import { MicrosoftStrategy } from './strategies/microsoft.strategy';
     AuthService,
     JwtStrategy,
     RTJwtStrategy,
-    GoogleStrategy,
-    GithubStrategy,
-    MicrosoftStrategy,
+    authProviderCheck('GOOGLE') ? GoogleStrategy : EmptyClassProvider,
+    authProviderCheck('GITHUB') ? GithubStrategy : EmptyClassProvider,
+    authProviderCheck('MICROSOFT') ? MicrosoftStrategy : EmptyClassProvider,
```

Comment:
I think, we do not need `EmptyClassProvider`
```suggestion
    ...(authProviderCheck('GOOGLE') ? [GoogleStrategy] : []),
    ...(authProviderCheck('GITHUB') ? [GithubStrategy] : []),
    ...(authProviderCheck('MICROSOFT') ? [MicrosoftStrategy] : []),
```